pkg/admission/validator/shoot.go

```diff
@@ -93,13 +93,18 @@ var (
 	workerConfigPath         = func(index int) *field.Path { return workersPath.Index(index).Child("providerConfig") }
 )
 
+type workerConfigContext struct {
+	workerConfig *apiskubevirt.WorkerConfig
+	dataVolumes  []core.DataVolume
```

Comment:
Note that `dataVolumes` can only contain non-root volumes. I believe you also need the root volume here, so you should add an additional field, `volume core.Volume`.

Reply:
I don't have to pass root volume after all, i just have to consider root disk. I will push the fix in one minute.

Reply:
You are right, thanks.